src/main/java/org/broadinstitute/hellbender/tools/walkers/mutect/SomaticReferenceConfidenceModel.java

```diff
@@ -0,0 +1,91 @@
+     */
+    SomaticReferenceConfidenceModel(final SampleList samples,
+                                    final SAMFileHeader header,
+                                    final int indelInformativeDepthIndelSize,
+                                    final SomaticGenotypingEngine genotypingEngine){
+        super(samples, header, indelInformativeDepthIndelSize, 0);
+        this.samples = samples;
+        this.genotypingEngine = genotypingEngine;
+    }
+
+        /**
+         * Calculate the genotype likelihoods for the sample in pileup for being hom-ref contrasted with being ref vs. alt
+         *
+         * @param ploidy target sample ploidy.
+         * @param pileup the read backed pileup containing the data we want to evaluate
+         * @param refBase the reference base at this pileup position
+         * @param qual the min base quality for a read in the pileup at the pileup position to be included in the calculation
+         * @param hqSoftClips running average data structure (can be null) to collect information about the number of high quality soft clips
+         * @return a RefVsAnyResult genotype call.
+         */
+    @Override
+    public ReferenceConfidenceResult calcGenotypeLikelihoodsOfRefVsAny(final int ploidy,
+                                                                       final ReadPileup pileup,
+                                                                       final byte refBase,
+                                                                       final byte qual,
+                                                                       final MathUtils.RunningAverage hqSoftClips,
+                                                                       final boolean readsWereRealigned) {
+
+        final SomaticRefVsAnyResult result = new SomaticRefVsAnyResult();
+        final Map<String, List<GATKRead>> perSampleReadMap = new HashMap<>();
+        perSampleReadMap.put(samples.getSample(0), pileup.getReads());
+        final ReadLikelihoods<Allele> readLikelihoods = new ReadLikelihoods<>(samples, new IndexedAlleleList<>(Arrays.asList(Allele.create(refBase,true), Allele.NON_REF_ALLELE)), perSampleReadMap);
+        final Iterator<PileupElement> pileupIter = pileup.iterator();
+        for (int i = 0; i < pileup.size(); i++) {
+            final PileupElement element = pileupIter.next();
+            final boolean isAlt = readsWereRealigned ? isAltAfterAssembly(element, refBase) : isAltBeforeAssembly(element, refBase);
+            final double nonRefLikelihood;
+            if (isAlt) {
+                nonRefLikelihood = QualityUtils.qualToProbLog10(qual);
+                result.nonRefDepth++;
+            } else {
+                nonRefLikelihood = QualityUtils.qualToErrorProbLog10(qual) + MathUtils.LOG10_ONE_THIRD;
+                result.refDepth++;
+            }
+            readLikelihoods.sampleMatrix(0).set(0, i, nonRefLikelihood);
```

Comment:
If you're calculating `nonRefLikelihood` why are you setting allele `0` and not `1` here?

Reply:
I guess I'm having trouble wrapping my head around the variable allele fraction.  There's something that surprises me about the asymmetry of your model.

1 Q10 alt and 758 refs is -2.8 against the NON-REF (if I change the calculations as you suggest)
1 ref and 818 alts is 2596 for the ALT

If the statistic is a log odds then I would expect the alt versus ref LOD to be the negative of the ref versus alt LOD but those nearly complementary pileups have drastically different LODs.  Either my assumption of symmetry (LOD of the flipped comparison being negative of the original) is wrong or there's something I don't understand about the priors.

Reply:
The -2.8 isn't really *against* the NON-REF, per se.  It's -2.8 *for* the NON-REF, that is to say, very weak evidence that the NON-REF allele might exist.

The somatic likelihoods model doesn't assign a LOD to the reference allele.  In fact, it doesn't *compare* the reference allele to the alt alleles directly.  Rather, it's the log odds of excluding each alt allele.  That is, your calculations mean that

* 1 alt and 758 refs: the likelihood of no variation (only the ref is present) is 10^2.8 times as great as the likelihood for both the ref and alt to be present
* 1 ref and 818 alts: the likelihood of no variation is 10^2596 times smaller than the likelihood for the ref and alt to be present.

In the case of 1 alt and 758 refs if we calculated the probability of just the alt allele and no ref we would also get something preposterous like 10^2596.

I could calculate the ref allele's LOD to restore symmetry if you think the M2 GVCF needs it.